websites/C/Craiyon/presence.ts

```diff
@@ -0,0 +1,60 @@
+const presence = new Presence({
+	clientId: "1015402986534608948",
+});
+
+let elapsedTimestamp: number = null,
+	oldPrompt: string = null;
+
+presence.on("UpdateData", () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/uCqmcTv.png",
+		},
+		{ pathname } = window.location;
+	switch (pathname) {
+		case "/": {
+			const input = document.querySelector<HTMLDivElement>("#prompt"),
+				container = document.querySelector(
+					".h-full.w-full > .relative > div > div"
+				);
+			presenceData.state = input.textContent ? `"${input.textContent}"` : "";
+			if (container.querySelector("svg.text-gray-300"))
+				presenceData.details = "Thinking of a prompt";
+			else if (
+				input.nextElementSibling
+					.querySelector("img")
+					.classList.contains("animate-wiggle")
+			) {
+				if (elapsedTimestamp === null) {
+					elapsedTimestamp = Date.now() / 1000;
+					oldPrompt = input.textContent;
+				}
+				presenceData.details = "Generating images";
+				presenceData.state = `"${oldPrompt}"`;
+			} else {
+				elapsedTimestamp = null;
```

Comment:
why?

Reply:
I do not quite understand what the question is asking about. Assuming you are asking about ```presenceData.state= `"${oldPrompt}"` ```, the purpose is so that the presence displays the original prompt for the images even if the user modified the prompt midway

Reply:
No elapsedTimestamp